numpy/core/src/multiarray/shape.c

```diff
@@ -87,7 +87,14 @@ PyArray_Resize(PyArrayObject *self, PyArray_Dims *newshape, int refcheck,
         }
 
         if (refcheck) {
+#ifdef PYPY_VERSION
+            PyErr_SetString(PyExc_ValueError,
+                    "cannot resize an array on PyPy."\
+                    "  Use the resize function");
```

Comment:
While nitpicking,the commit message could be neater ;).

Anyway, I really like this development.... Wonder if that MaskedarrayFutureWarning (and the refcount check to make sure its not too noisy) isn't the reason for half of those errors....